src/attr.rs

```diff
@@ -136,8 +162,8 @@ impl<'a> AttributeList<'a> {
     }
 
     /// Unconditionally set an attribute to a value. If the attribute already
-    /// exists in the set will update its value, otherwise will add a new
-    /// attribute to the set.
+    /// exists in the list will update its value, otherwise will add a new
+    /// attribute to the list.
```

Comment:
I think this sounds better: `If the attribute already exists in the list, update its value, otherwise add a new attribute to the list.`.